components/ReduceOneItem.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useEffect, useState } from 'react';
+import cookie from 'js-cookie';
+
+type Item = {
+  id: string;
+  name: string;
+  img: string;
+  amount: number;
+  price: number;
+  info: string;
+};
+
+type Cart = {
+  id: string;
+  name: string;
+  img: string;
+  amount: number;
+  price: number;
+  info: string;
+};
+
+type Products = {
+  id: string;
+  name: string;
+  img: string;
+  amount: number;
+  price: number;
+  info: string;
+};
+type Props = { item: Item; cart: Cart[]; products: Products[] };
+export default function ReduceOneItem(props: Props) {
+  // const [itemsInCart, setItemsInCart] = useState(props.itemsInCart || []);
+
+  // useEffect(() => {
+  //   cookie.set('cart', props.itemsInCart);
+  // }, [props.itemsInCart]);
+
+  function reduceOne(id) {
+    const newCart = props.cart.map((item) => {
+      if (item.id === id) {
+        if (item.amount === 1) {
+          alert('Use remove button!');
+          return item;
+        } else {
+          const prodPrice = props.products.find((prod) => prod.id === id);
+          return {
+            ...item,
+            amount: +item.amount - 1,
+            price: +prodPrice.price * (+item.amount - 1),
```

Comment:
It seems like from your TypeScript types above that you shouldn't need to cast to numbers by using the `+` symbol on either of these lines.